modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheObjectUtils.java

```diff
@@ -201,8 +203,10 @@ public static Object unwrapBinary(
             return unwrapKnownCollection(ctx, (Collection<Object>)o, keepBinary, cpy);
         else if (BinaryUtils.knownMap(o))
             return unwrapBinariesIfNeeded(ctx, (Map<Object, Object>)o, keepBinary, cpy);
-        else if (o instanceof Object[])
+        else if (o instanceof Object[] && !BinaryArray.useBinaryArrays())
```

Comment:
Do we really need `!BinaryArray.useBinaryArrays()` condition here?

Reply:
This method creates new object on every invocation:
1. It's not good from performance perspective.
2. There are tests that check same object(by reference) returned so without this condition tests fails.